cli/azd/cmd/show.go

```diff
@@ -165,7 +178,82 @@ func (s *showAction) Run(ctx context.Context) (*actions.ActionResult, error) {
 		}
 	}
 
-	return nil, s.formatter.Format(res, s.writer, nil)
+	if s.formatter.Kind() == output.JsonFormat {
+		return nil, s.formatter.Format(res, s.writer, nil)
+	}
+
+	appEnvironments, err := s.envManager.List(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	uxEnvironments := make([]*ux.ShowEnvironment, len(appEnvironments))
+	for index, environment := range appEnvironments {
+		uxEnvironments[index] = &ux.ShowEnvironment{
+			Name:      environment.Name,
+			IsCurrent: environment.Name == environmentName,
+			IsRemote:  !environment.HasLocal && environment.HasRemote,
+		}
+	}
+
+	uxServices := make([]*ux.ShowService, len(res.Services))
+	var index int
+	for serviceName, service := range res.Services {
+		uxServices[index] = &ux.ShowService{
+			Name:      serviceName,
+			IngresUrl: service.IngresUrl,
+		}
+		index++
+	}
+
+	s.console.MessageUxItem(ctx, &ux.Show{
+		AppName:         s.azdCtx.GetDefaultProjectName(),
+		Services:        uxServices,
+		Environments:    uxEnvironments,
+		AzurePortalLink: azurePortalLink(subId, rgName),
+	})
+
+	return nil, nil
+}
+
+func (s *showAction) serviceEndpoint(
+	ctx context.Context, subId string, serviceConfig *project.ServiceConfig, env *environment.Environment) string {
+	resourceManager, err := s.lazyResourceManager.GetValue()
+	if err != nil {
+		log.Printf("error: getting lazy target-resource. Endpoints will be empty: %v", err)
+		return ""
+	}
+	targetResource, err := resourceManager.GetTargetResource(ctx, subId, serviceConfig)
+	if err != nil {
+		log.Printf("error: getting target-resource. Endpoints will be empty: %v", err)
+		return ""
+	}
+
+	serviceManager, err := s.lazyServiceManager.GetValue()
+	if err != nil {
+		log.Printf("error: getting lazy service manager. Endpoints will be empty: %v", err)
+		return ""
+	}
+	st, err := serviceManager.GetServiceTarget(ctx, serviceConfig)
+	if err != nil {
+		log.Printf("error: getting service target. Endpoints will be empty: %v", err)
+		return ""
+	}
+	endpoints, err := st.Endpoints(ctx, serviceConfig, targetResource)
+	if err != nil {
+		log.Printf("error: getting service endpoints. Endpoints might be empty: %v", err)
+	}
+
+	overriddenEndpoints := project.OverriddenEndpoints(ctx, serviceConfig, env)
+	if len(overriddenEndpoints) > 0 {
+		endpoints = overriddenEndpoints
+	}
+
+	if len(endpoints) == 0 {
+		return ""
+	}
+
+	return endpoints[0]
```

Comment:
Do we only show one? Feels counter intuitive since there can be multiple

Reply:
That's what we do for the current output from provision (pick only one). I got it from there.
I honestly haven't seen a case where there are multiple ingress-urls for a service.

However, if multiple endpoints are possible, we need a fix here and also on provision. I would have that as another PR